lib/bap_image/objectFile.mli

```diff
@@ -0,0 +1,50 @@
+(** LLVM object file *)
+
+open Core_kernel.Std
+
+type t
+
+external create : Bigstring.t -> t Or_error.t =
+  "object_file_create_stub"
+
+type archType =
```

Comment:
What is the rationale to have intermediate type? You have already the `arch` type with the same set of constructors exposed to you.